src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTableUI.java

```diff
@@ -2091,20 +2098,22 @@ private void paintCells(Graphics g, int rMin, int rMax, int cMin, int cMax) {
         } else {
             for(int row = rMin; row <= rMax; row++) {
                 cellRect = table.getCellRect(row, cMin, false);
-                aColumn = cm.getColumn(cMin);
+                cellRect.x = getXPosition(cMax);
+                aColumn = cm.getColumn(cMax);
                 if (aColumn != draggedColumn) {
                     columnWidth = aColumn.getWidth();
                     cellRect.width = columnWidth - columnMargin;
-                    paintCell(g, cellRect, row, cMin);
+                    paintCell(g, cellRect, row, cMax);
```

Comment:
Is this code snippet required? It was not needed for LTR..
Guess we do the check in l2113 where we do paintCell for the same..

Reply:
Yeah, it maybe to ensure minimum one column is painted/visible during Table drag i.e., meaning when table is shrink to min. But not exactly sure why one column is painted first and then the remaining.

Reply:
Ok..But why it is needed for RTL and not for LTR?

Reply:
My guess is for RTL the first column might not start from extreme left, as we drag it moves towards the Left corner, this might be for that, still not sure, will verify and come back on this.

Reply:
You are right, its not required and couldn't find any reason why it was added.